mangaki/mangaki/management/commands/compare.py

```diff
@@ -60,25 +101,53 @@ def load_dataset(self, dataset_name):
         )
         self.anonymized = dataset.anonymized
 
+    def compute_metrics(self, model, y_pred, i_test):
+        results = {}
+        for metric in self.evaluation_metrics:
+            compute_method = getattr(model, 'compute_{}'.format(metric))
+            results[metric] = compute_method(y_pred, self.anonymized.y[i_test])
+            if model.verbose:
+                logger.debug('{} ({}) {:f}'.format(metric,
+                                                   i_test,
+                                                   results[metric]))
+
+        return results
+
     def compare_models(self):
         k_fold = ShuffleSplit(n_splits=NB_SPLIT)
-        rmse_values = defaultdict(list)
+        metrics = defaultdict(lambda: defaultdict(list))
+
+        pass_index = 0
         for i_train, i_test in k_fold.split(self.anonymized.X):
             for algo in self.algos:
-                model = algo()
-                logger.info(model.get_shortname())
+                model = algo.make_instance()
+                logger.info('[{0} {1}-folding] pass={2}/{1}'.format(model.get_shortname(), NB_SPLIT, pass_index + 1))
                 model.set_parameters(self.anonymized.nb_users, self.anonymized.nb_works)
                 model.fit(self.anonymized.X[i_train], self.anonymized.y[i_train])
                 y_pred = model.predict(self.anonymized.X[i_test])
-                rmse = model.compute_rmse(y_pred, self.anonymized.y[i_test])
                 if model.verbose:
-                    logger.debug('Predicted: %s' % y_pred[:5])
-                    logger.debug('Was: %s' % self.anonymized.y[i_test][:5])
-                logger.debug('RMSE %f' % rmse)
-                rmse_values[model.get_shortname()].append(rmse)
+                    logger.info('Predicted: %s' % y_pred[:5])
+                    logger.info('Was: %s' % self.anonymized.y[i_test][:5])
+
+                metrics_values = self.compute_metrics(model, y_pred, i_test)
+                for metric, value in metrics_values.items():
+                    metrics[metric][model.get_shortname()].append(value)
+
+            pass_index += 1
+
         logger.info('Final results')
-        for algo_name in rmse_values:
-            logger.info('%s: RMSE = %f' % (algo_name, np.mean(rmse_values[algo_name])))
+        for metric_name, algos in metrics.items():
+            logger.info('Evaluation metric with {} results:'.format(metric_name))
```

Comment:
No need to write `metric` or `results`. `Evaluation of RMSE:` is enough :) (please use capitals). Thanks ^^